ecs-agent/tmds/utils/netconfig/netconfig_linux.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build !windows
+// +build !windows
+
+package netconfig
+
+import (
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger/field"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/utils/netlinkwrapper"
+
+	"github.com/vishvananda/netlink"
+)
+
+// DefaultNetInterfaceName returns the device name of the first default network interface
+// available on the instance. If none exist, an empty string and nil will be returned.
+func DefaultNetInterfaceName(netlinkClient netlinkwrapper.NetLink) (string, error) {
+	routes, err := netlinkClient.RouteList(nil, netlink.FAMILY_ALL)
+	if err != nil {
+		return "", err
+	}
+
+	// Iterate over all routes
+	for _, route := range routes {
+		if route.Gw == nil {
```

Comment:
FYI - learn this from the 3rd party lib - https://github.com/tailscale/tailscale/blob/main/net/netmon/interfaces_linux.go#L168-L193